src/mgr/DaemonServer.cc

```diff
@@ -498,7 +498,7 @@ bool DaemonServer::handle_command(MCommand *m)
     int acting_primary = -1;
     entity_inst_t inst;
     cluster_state.with_osdmap([&](const OSDMap& osdmap) {
-	osdmap.pg_to_acting_osds(pgid, nullptr, &acting_primary);
+	acting_primary = osdmap.get_pg_acting_primary(pgid);
```

Comment:
`osdmap.pg_to_acting_osds()` segfaults when compiled using `-O0` where the returned value of this method is not optimized out, so the compiled does translate `return acting->size()` to deadly instructions.